libs/settings/include/settings/detail/setting_list.hpp

```diff
@@ -0,0 +1,111 @@
+#pragma once
```

Comment:
this file is usable more generally; namespacing it as a detail of settings may need a rethink. Could it find a home together with the EndsWith string helper that you recently wrote?

Unrelatedly, setting_list.hpp as a name may be confusing, especially given that setting_collection.hpp exists and contains very different functionality.

Reply:
Yes it might be useful. I would suggest that we find that use case first and then generalise it. I am happy for filename suggestions but this seems fine to me

Reply:
csv_string_helpers.hpp seems truthful enough